quit/core.py

```diff
@@ -413,17 +423,74 @@ def getFileReferenceAndContext(self, blob, commit):
             content = commit.node(path=name).content
             graphUri = self._graphconfigs.get(commit.id).getgraphuriforfile(name)
             graph = Graph(identifier=URIRef(graphUri))
-            graph.parse(data=content, format='nt')
-            quitWorkingData = (FileReference(name, content), graph)
+            parserGraph = ntriples.W3CNTriplesParser(ntriples.NTGraphSink(graph))
+            source = rdflib.parser.create_input_source(data=content)
+            parserGraph.parse(source.getCharacterStream())
+            nameMap = {v: k for k, v in parserGraph._bnode_ids.items()}
+            quitWorkingData = (FileReference(name, content), graph, nameMap)
             self._blobs.set(blob, quitWorkingData)
             return quitWorkingData
         return self._blobs.get(blob)
 
+    def _replaceLabledBlankNodes(self, parsedQuery, parent_commit_ref):
+        """Replaces blanknodes in parsedQuery with Blanknodes that have the same label in the graph.nt
+        E.g.  We have a Graph with the content: '_:a <urn:pred> _:b'
+        A BNode('a') found in parsedQuery would be replaced by the blanknode _:a found in the graph.nt.
+        That way, updates can pass Blanknodes as instances and do not have to work on string representations.
+        """
+        def replaceBlankNode(parsedQuery, nameMap):
```

Comment:
Very very big: three for loops and four ifs, try to split in multiple functions and maybe use [list comprehension](https://www.w3schools.com/python/python_lists_comprehension.asp) for more conciseness.